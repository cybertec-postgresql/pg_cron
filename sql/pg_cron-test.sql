-- Check the update procedure
CREATE EXTENSION pg_cron VERSION '1.0';
SELECT extversion FROM pg_extension WHERE extname='pg_cron';
ALTER EXTENSION pg_cron UPDATE TO '1.1';
SELECT extversion FROM pg_extension WHERE extname='pg_cron';
ALTER EXTENSION pg_cron UPDATE TO '1.2';
SELECT extversion FROM pg_extension WHERE extname='pg_cron';

SELECT cron.schedule('0 10 * * *', 'VACUUM');
SELECT cron.schedule('@daily', 'SELECT pg_sleep(1');
SELECT cron.schedule('@interval(00:00:10)', 'SELECT pg_sleep(1');
SELECT cron.schedule('00:00:22'::interval, 'SELECT pg_sleep(1');
SELECT cron.schedule('8 hours'::interval, 'SELECT pg_sleep(1');
SELECT cron.schedule('1 seconds'::interval, 'SELECT pg_sleep(1');
SELECT cron.schedule('2 days'::interval, 'SELECT pg_sleep(1');
SELECT cron.schedule('@interval(02 00:30:00)', 'SELECT pg_sleep(1');
SELECT cron.schedule('2 days 12 minutes'::interval, 'SELECT pg_sleep(1');
SELECT cron.schedule('00:07:00'::interval, 'SELECT pg_sleep(1');
SELECT cron.schedule('01:00:00'::interval, 'SELECT pg_sleep(1');

-- Stop scheduling a job
DO $$
BEGIN
    FOR i IN 1..11 LOOP
        PERFORM cron.unschedule(i);
    END LOOP;
END$$;

DROP EXTENSION pg_cron;