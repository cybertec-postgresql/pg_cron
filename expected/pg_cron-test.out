-- Check the update procedure
CREATE EXTENSION pg_cron VERSION '1.0';
SELECT extversion FROM pg_extension WHERE extname='pg_cron';
 extversion 
------------
 1.0
(1 row)

ALTER EXTENSION pg_cron UPDATE TO '1.1';
SELECT extversion FROM pg_extension WHERE extname='pg_cron';
 extversion 
------------
 1.1
(1 row)

ALTER EXTENSION pg_cron UPDATE TO '1.2';
SELECT extversion FROM pg_extension WHERE extname='pg_cron';
 extversion 
------------
 1.2
(1 row)

SELECT cron.schedule('0 10 * * *', 'VACUUM');
 schedule 
----------
        1
(1 row)

SELECT cron.schedule('@daily', 'SELECT pg_sleep(1');
 schedule 
----------
        2
(1 row)

SELECT cron.schedule('@interval(00:00:10)', 'SELECT pg_sleep(1');
 schedule 
----------
        3
(1 row)

SELECT cron.schedule('00:00:22'::interval, 'SELECT pg_sleep(1');
 schedule 
----------
        4
(1 row)

SELECT cron.schedule('8 hours'::interval, 'SELECT pg_sleep(1');
 schedule 
----------
        5
(1 row)

SELECT cron.schedule('1 seconds'::interval, 'SELECT pg_sleep(1');
 schedule 
----------
        6
(1 row)

SELECT cron.schedule('2 days'::interval, 'SELECT pg_sleep(1');
 schedule 
----------
        7
(1 row)

SELECT cron.schedule('@interval(02 00:30:00)', 'SELECT pg_sleep(1');
 schedule 
----------
        8
(1 row)

SELECT cron.schedule('2 days 12 minutes'::interval, 'SELECT pg_sleep(1');
 schedule 
----------
        9
(1 row)

SELECT cron.schedule('00:07:00'::interval, 'SELECT pg_sleep(1');
 schedule 
----------
       10
(1 row)

SELECT cron.schedule('01:00:00'::interval, 'SELECT pg_sleep(1');
 schedule 
----------
       11
(1 row)

-- Stop scheduling a job
DO $$
BEGIN
    FOR i IN 1..11 LOOP
        PERFORM cron.unschedule(i);
    END LOOP;
END$$;
DROP EXTENSION pg_cron;
